src/samples/design_patterns/Makefile

```diff
@@ -0,0 +1,2 @@
+sample-$(DESIGN_PATTERN_SAMPLES) += stream_sample
+sample-stream_sample-$(DESIGN_PATTERN_SAMPLES) := stream_sample.c
```

Comment:
that's okay, but then change Kconfig to reflect that. There you have both knobs.